### PR TITLE
Metric resolution

### PR DESCRIPTION
**Pull request by cpennington**
@alawibaba @macdiesel @doctoryes: This allows us to configure how many metrics to log, and also separates the file logs based on whether the locusts are hatching, or running.

### CODE REVIEW

File: helpers/raw_logs.py [hunk 1]
```diff
@@ -36,23 +44,46 @@ def _open_log_file(self):
...
+    def on_hatch_complete(self, **kwargs):
+        self.hatching = False
+        self._metric_counter = 0
+        self._close_log_file()
+
+    def _skip_entry(self):
+        """
+        Return True if the logger should skip this entry.
+        """
+        self._metric_counter += METRIC_RESOLUTION
```
Comment (doctoryes) [inline helpers/raw_logs.py]: If the first thing done when calling `_skip_entry` is adding the `METRIC_RESOLUTION`, how can `_metric_counter` ever become negative?

Comment (cpennington) [inline helpers/raw_logs.py]: It won't become negative. It starts at 0, and gets `METRIC_RESOLUTION` added whenever we try to log a metric. If it gets greater than 1, then we *actually* log a metric, and then subtract 1. So, if `METRIC_RESOLUTION` is `0.3`, then we'd see:

    0.3, 0.6, 0.9, *1.2*, 0.5, 0.8, *1.1*, 0.4, 0.7, *1.0*, 0.3, ...

Comment (doctoryes) [inline helpers/raw_logs.py]: Ah - I was thinking that `METRIC_RESOLUTION` would be greater than 1. Perhaps adding a comment and/or a limit on the value when reading it above would be helpful.

Comment (cpennington) [inline helpers/raw_logs.py]: Good call. Added.

**Comment by doctoryes** [comment]
:+1:

**Pull request merged**